### ISSUE TITLE
Attention weight in CrossAttnDownBlock3D is not trained?

### ISSUE THREAD

**Issue by zhangfuyang** [showlab/Tune-A-Video#18]
Hi, Congrats on this awesome work!

One question. For this [line](https://github.com/showlab/Tune-A-Video/blob/main/tuneavideo/models/unet_blocks.py#L300), it seems the gradient is not going through. 

I'm not familiar with torch.utils.checkpoint.checkpoint. Is this something to re-calculate the gradient during the backward pass while using no_grad() during the forward time?

However, even so, I didn't observe the weight change after one iteration of training. Specifically, `unet.down_blocks[0].attentions[0].transformer_blocks[0].attn1.to_q.weight` is not changed.

Is this correct? or I missed something?

**Comment by lealaxy** [comment]
same question.
